sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala

```diff
@@ -129,6 +129,14 @@ private[spark] object HiveUtils extends Logging {
     .toSequence
     .createWithDefault(Nil)
 
+  val HIVE_METASTORE_SHARE_HADOOPCLASSES =
+    SQLConfigBuilder("spark.sql.hive.metastore.shareHadoopClasses")
+      .doc("When set to true, the classloader used to load Hive metastore client will " +
+        "not explicitly load Hadoop classes. Instead, Hadoop classes loaded by the classloader " +
+        "that loads Spark will be used.")
+      .booleanConf
+      .createWithDefault(false)
```

Comment:
@marmbrus I am setting this conf to false by default, which changes the behavior (our master always share hadoop clsses).